### PR TITLE
Add server command for oidc user history

### PR DESCRIPTION
**Pull request by abretaud**
Adding a server command to fill our influxdb with the evolution of the number of OIDC accounts since the opening of the server.

Strangely, I tried to name the function `server_users-with-oidc`, but it ran `server_users` at the same time, not sure if it's intended.

### CODE REVIEW

File: parts/29-server.sh [hunk 1]
```diff
@@ -38,6 +38,38 @@ server_users() { ## : Count of different classifications of users
 	EOF
 }
 
+server_oidc() { ## : How many users logged in with OIDC
+	meta <<-EOF
+		ADDED: 21
```
Comment (hexylena) [inline parts/29-server.sh:43]:
```suggestion
		ADDED: 21
		AUTHORS: abretaud
```

File: parts/29-server.sh [hunk 2]
```diff
@@ -38,6 +38,38 @@ server_users() { ## : Count of different classifications of users
 	EOF
 }
 
+server_oidc() { ## : How many users logged in with OIDC
```
Comment (hexylena) [inline parts/29-server.sh:41]:
```suggestion
server_oidc() { ##? [--op=<=,!=,...>] [--date=<yyyy-mm-dd>] : How many users logged in with OIDC
```

File: parts/29-server.sh [hunk 3]
```diff
@@ -38,6 +38,38 @@ server_users() { ## : Count of different classifications of users
...
+	meta <<-EOF
+		ADDED: 21
+	EOF
+	handle_help "$@" <<-EOF
+	EOF
+
+	op="="
+	if (( $# > 1 )); then
+		op="$2"
+	fi
```
Comment (hexylena) [inline parts/29-server.sh:51]:
```suggestion
	op="="
	if [[ -z "$arg_op" ]]; then
		op="$arg_op"
	fi
```

There's a new "default value" syntax that can be used but...accepting strings like `=` is going to completely break that parsing so, we just leave this the old way.

File: parts/29-server.sh [hunk 4]
```diff
@@ -38,6 +38,38 @@ server_users() { ## : Count of different classifications of users
...
+
+	date_filter=""
+	if (( $# > 0 )); then
+		date_filter="WHERE date_trunc('day', galaxy_user.create_time AT TIME ZONE 'UTC') $op '$1'::date"
+	fi
```
Comment (hexylena) [inline parts/29-server.sh:56]:
```suggestion
	date_filter=""
	if [[ -z "$arg_date" ]] then
		date_filter="WHERE date_trunc('day', galaxy_user.create_time AT TIME ZONE 'UTC') $op '$arg_date'::date"
	fi
```

**Review by hexylena — COMMENTED** [review]
Looks good overall! Could you mention it in the changelog.md for me?

**Comment by abretaud** [comment]
Thanks for the review :)

> Looks good overall! Could you mention it in the changelog.md for me?

I think I did

I'm looking into adding something similar for disk-usage, maybe I can push to this branch before merging

**Comment by abretaud** [comment]
Ok, fixed args stuff (wap doesn't like `=` in possible values for --op option), now it's working with these command:

```
gxadmin server oidc
gxadmin server oidc --date=2021-01-01
gxadmin server oidc --op "<=" --date=2021-01-01
```

But, I'd like to use it with the slurp commands, and it fails, like this for example:

```
blabla@bloblo:~$ gxadmin meta slurp-upto 2021-01-01 server_oidc
Error: more positional arguments than should be possible
Error: more positional arguments than should be possible

meta oidc - 

**SYNOPSIS**
```

Running `gxadmin meta slurp-upto --date=2021-01-01 server_oidc` removed the first error, but I can pass the --op option, and its default value is enough to trigger the second error

Any idea how to fix that @hexylena?

**Comment by hexylena** [comment]
aye I'll have a look!

**Comment by hexylena** [comment]
> Strangely, I tried to name the function server_users-with-oidc, but it ran server_users at the same time, not sure if it's intended.

I cannot reproduce this? 

<details>

```
12:25:21|(pr-124‽) [user@w-galaxy:~/arbeit/galaxy/gxadmin]$ git diff | cat
diff --git a/parts/29-server.sh b/parts/29-server.sh
index 7b7624d..b75c944 100644
--- a/parts/29-server.sh
+++ b/parts/29-server.sh
@@ -38,7 +38,7 @@ server_users() { ## : Count of different classifications of users
 	EOF
 }
 
-server_oidc() { ##? [--op=<...>] [--date=<yyyy-mm-dd>] : How many users logged in with OIDC
+server_users-with-oidc() { ##? [--op=<...>] [--date=<yyyy-mm-dd>] : How many users logged in with OIDC
 	meta <<-EOF
 		ADDED: 21
 		AUTHORS: abretaud
12:25:22|(pr-124‽) [user@w-galaxy:~/arbeit/galaxy/gxadmin]$ make
make: Nothing to be done for 'defaut'.
12:25:25|(pr-124‽) [user@w-galaxy:~/arbeit/galaxy/gxadmin]$ ./gxadmin echoserver users-with-oidc
SELECT
provider, count(distinct user_id)
FROM
oidc_user_authnz_tokens
JOIN
galaxy_user ON oidc_user_authnz_tokens.user_id = galaxy_user.id

GROUP BY provider
12:25:26|(pr-124‽) [user@w-galaxy:~/arbeit/galaxy/gxadmin]$ ./gxadmin server users-with-oidc    
 provider | count 
----------+-------
(0 rows)
```
</details>

Edit: Ahh you mean in the slurp-upto. Indeed, poor matching function was written.

I've pushed some changes which should fix it by:
- using the new argument parser everywhere in server
- using it also in the slurp-upto
- changing the way slurp-upto was done, rather than checking if the names were equal (which only checked prefixes :|) it just uses the passed list of names. This is obviously a much better solution, I'm not sure why I didn't do this originally.

**Comment by abretaud** [comment]
Tested and approved! I've added a slurpable disk-usage function in the same way

**Comment by hexylena** [comment]
fantastic work @abretaud !

**Pull request merged**